### PR TITLE
Spike mes 2067 show fields based on activity code

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
MES 2067 is a tech spike to prove that we can map the data from the validation spreadsheet to a data structure and use that data to control displaying form fields (or not) and validation of those fields (or not), plus changing options in drop downs etc.

This behaviour mapping will be extended next sprint for all form controls on the office page, and for all test outcomes. 

It also relies on testOutcome (activity code) being populated so at the moment defaults to a known value.

### CODE REVIEW

File: src/providers/outcome-behaviour-map/outcome-behaviour-map.model.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export interface FieldDetail {
+  display: string;
```
Comment (ammarhaiderbjss) [inline src/providers/outcome-behaviour-map/outcome-behaviour-map.model.ts]: can we make this a string union type if there are just 3 values? i.e.
`display: 'Y' | 'N' | 'A';`

File: src/modules/tests/__tests__/tests.selector.spec.ts [hunk 2]
```diff
@@ -156,8 +156,8 @@ describe('testsSelector', () => {
     });
     it('should return mes-red class for non listed activity code 86', () => {
       testState.activityCode = '86';
-      const result = getTestOutcome(testState);
-      expect(result).toBe('Terminated');
+      const result = getTestOutcomeClass(testState);
+      expect(result).toBe('mes-red');
     });
   });
```
Comment (ammarhaiderbjss) [inline src/modules/tests/__tests__/tests.selector.spec.ts]: Unit test for `getShowMeQuestionOptions`?

File: src/pages/office/office.ts [hunk 3]
```diff
@@ -198,6 +222,9 @@ export class OfficePage extends BasePageComponent {
         select(getVehicleChecks),
         select(getShowMeQuestion),
       ),
+      showMeQuestionOptions$: currentTest$.pipe(
+        map(() => getShowMeQuestionOptions(this.showMeQuestions, '4', this.outcomeBehaviourProvider)),
```
Comment (ammarhaiderbjss) [inline src/pages/office/office.ts]: Perhaps a TODO comment here to replace the hardcoded outcome?

File: src/providers/outcome-behaviour-map/outcome-behaviour-map.model.ts (hunk 1 above)
Comment (rossfauldsbjss) [reply src/providers/outcome-behaviour-map/outcome-behaviour-map.model.ts]: Done.

File: src/modules/tests/__tests__/tests.selector.spec.ts (hunk 2 above)
Comment (rossfauldsbjss) [reply src/modules/tests/__tests__/tests.selector.spec.ts]: oh yes. good spot. will do that now.

File: src/pages/office/office.ts (hunk 3 above)
Comment (rossfauldsbjss) [reply src/pages/office/office.ts]: definite TODO needed. Changing now.

Comment (rossfauldsbjss) [reply src/pages/office/office.ts]: Have wired this up to take outcome from state, so no TODO needed

File: src/providers/outcome-behaviour-map/outcome-behaviour-map.model.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export interface FieldDetail {
+  display: 'A' | 'Y' | 'N';
```
Comment (johnmorleybjss) [inline src/providers/outcome-behaviour-map/outcome-behaviour-map.model.ts]: As discussed, enum around this would be good.

**Pull request merged**